frontend/test/specs/event/eventControllerSpec.js

```diff
@@ -0,0 +1,170 @@
+        'text': 'Text',
+        'local': 'Local',
+        'photo_url': null,
+        'start_time': date,
+        'end_time': date_next_month,
+        'institution_key': other_institution.key
+    };
+
+    var requestEvent = {
+        events: [ event, other_event ],
+        next: false
+    };
+
+    var requestEventInst = {
+        events: [ event ],
+        next: false
+    };
+
+    var GET_EVENTS_URI = '/api/events?page=0&limit=5';
+    var GET_EVENTS_INST_URI = '/api/institutions/'+institution.key+'/events?page=0&limit=5';
+
+    beforeEach(module('app'));
+
+    beforeEach(inject(function ($controller, $httpBackend, AuthService,
+        $rootScope, EventService, MessageService, $mdDialog, $state) {
+        scope = $rootScope.$new();
+        httpBackend = $httpBackend;
+        rootScope = $rootScope;
+        eventService = EventService;
+        messageService = MessageService;
+        mdDialog = $mdDialog;
+        state = $state;
+        AuthService.login(user);
+
+        $httpBackend.expect('GET', GET_EVENTS_URI).respond(requestEvent);
+        $httpBackend.expect('GET', GET_EVENTS_INST_URI).respond(requestEventInst);
+
+        spyOn(Utils, 'setScrollListener').and.callFake(function () {
+            return {
+                then: function (callback) {
+                    return callback();
+                }
+            };
+        });
+
+        createCtrl = function(){
+            return $controller('EventController', {
+                scope: scope,
+                $rootScope: rootScope,
+                eventService: eventService,
+                messageService: messageService,
+                mdDialog: mdDialog
+            });
+        }
+
+        state.params.posts = [];
+        eventCtrl = createCtrl();
+        eventCtrl.showImage = true;
+        eventCtrl.events = [];
+    }));
+
+    describe('main()', function() {
+
+        it('should call functions', function() {
+            eventCtrl = createCtrl();
+            expect(eventCtrl.posts.length).toBe(0);
+        });
+    });
+
+    describe('loadMoreEvents()', function() {
+
+        beforeEach(function () {
+
+            spyOn(eventService, 'getEvents').and.callFake(function () {
+                return {
+                    then: function (callback) {
+                        return callback(requestEvent);
+                    }
+                };
+            });
+
+            spyOn(eventService, 'getInstEvents').and.callFake(function () {
+                return {
+                    then: function (callback) {
+                        return callback(requestEventInst);
+                    }
+                };
+            });
+        });
+
+        it('should call getEvents', function(done) {
+            expect(eventCtrl.events.length).toBe(0);
+            var promise = eventCtrl.loadMoreEvents();
+            promise.should.be.fulfilled.then(function() {
+                expect(eventCtrl.events.length).toBe(2);
```

Comment:
Check if this promise is actually being executed, I have already arranged tests where the same call (promise.should.be.fulfilled) existed and they were never executed and therefore always passed.